syntax.php

```diff
@@ -96,16 +83,22 @@ function handle($match, $state, $pos, Doku_Handler $handler) {
      */
     function render($mode, Doku_Renderer $renderer, $data) {
         global $lang;
+	global $ID;
+	global $INFO;
 
-        if ($mode == 'xhtml') {
+	if ($mode == 'xhtml') {
             $renderer->info['cache'] = false;
 
             @require_once(DOKU_INC.'inc/fulltext.php');
-            $backlinks = ft_backlinks($data[0]);
+	    $page_id = $data[0];
+	    if($page_id == '.') {
+		$page_id = $INFO['id'];
+	    }
```

Comment:
moving this here makes the check specific to xhtml mode and not any others like it did before